pkg/server/server.go

```diff
@@ -95,7 +88,12 @@ func NewApplicationsHandler(ctx context.Context, cfg *ApplicationConfig, opts ..
 }
 
 func (s *applicationServer) ListApplications(ctx context.Context, msg *pb.ListApplicationsRequest) (*pb.ListApplicationsResponse, error) {
-	apps, err := s.app.Kube.GetApplications(ctx, msg.GetNamespace())
+	kubeService, kubeError := s.appFactory.GetKubeService()
```

Comment:
Minor: I do think that we don't need this on a per call basis.  It might be safer though.  The code is already here.

Reply:
Talked f2f and we want to limit the scope of the refactoring.  This is good.